src/cast_core/test/CrudActorYamlTests.yml

```diff
@@ -651,13 +654,15 @@ Tests:
         Count: 1
     AssertNumTransactionsCommitted: 1
 
-  - Description: WithTransaction, BulkWrite insert, delete, then re-insert without OnSession on bulkWrite
+  - Description: WithTransaction, BulkWrite insert, delete, then re-insert without session on bulkWrite
     Operations:
       - OperationName: withTransaction
         OperationCommand:
+          OnSession: true
           OperationsInTransaction:
           - OperationName: bulkWrite
             OperationCommand:
+              OnSession: false
```

Comment:
What happened in the old code if `OnSession` alternated between true and false at this level?

Reply:
The inner `OnSession` (under `bulkWrite`) would override the outer one (under `withTransaction`). In the old test YAML code, `OnSession` under `withTransaction` wasn't set so `addOperation()` would assume that it's false, and `bulkWrite` would inherit this false value for `OnSession` from `withTransaction`.

Reply:
Yuck!

Reply:
Yeah all this implicit false and implicit value inheritance from outer scope is a bit weird 😂